CrowServe/CrowServe.h

```diff
@@ -65,9 +71,53 @@ class Socket {
         deinit_sockets();
     };
 
-
     bool try_init();
-    void send();
+
+    template <typename Handler>
+    void listen(const Handler& handler) { // todo - include cv for cancel
```

Comment:
We don't need cv. Could just be bool m_running that we set to false when we stop. Or can also use std::stop_token as first arg, and jthread mostly handles the rest. Most probably want to call request_stop followed with join before destroying other things in destructor.